### PR TITLE
Fine-tuned the style of contribution grid, which is exactly the same as github

### PR DESCRIPTION
**Pull request by xiandanin**
Added __rounded corners__, __border__, __corrected colors__, and the parameters are completely referenced to the latest github, making it closer to the style of github.

image

### CODE REVIEW

File: __tests__/draw.test.js [hunk 1]
```diff
@@ -14,7 +16,13 @@ test("draws contributions correctly", () => {
     username: "sallar",
     footerText: "test",
     themeName: "__test__",
-    fontFace: "PlexMono"
+    fontFace: "PlexMono",
+    scaleFactor: 2
   });
   expect(canvas.toDataURL()).toMatchSnapshot();
+
+  /*
```
Comment (sallar) [inline __tests__/draw.test.js:24]: this comment can be removed

File: __tests__/draw.test.js [hunk 2]
```diff
@@ -2,7 +2,9 @@ import path from "path";
 import mockData from "./mock.json";
 import { drawContributions } from "../src";
 import { createCanvas, registerFont } from "canvas";
+
 const canvas = createCanvas(200, 200);
+import fs from "fs";
```
Comment (sallar) [inline __tests__/draw.test.js:7]: unnecessary import

File: __tests__/draw.test.js [hunk 3]
```diff
@@ -14,7 +16,13 @@ test("draws contributions correctly", () => {
     username: "sallar",
     footerText: "test",
     themeName: "__test__",
-    fontFace: "PlexMono"
+    fontFace: "PlexMono",
+    scaleFactor: 2
```
Comment (sallar) [inline __tests__/draw.test.js:20]: this should be dynamic

File: src/themes.js [hunk 4]
```diff
@@ -1,22 +1,24 @@
...
   __test__: {
     background: "#ffffff",
-    text: "#ffffff",
-    meta: "#ffffff",
-    grade4: "#196127",
-    grade3: "#239a3b",
-    grade2: "#7bc96f",
-    grade1: "#c6e48b",
+    text: "#24292e",
+    meta: "#767676",
```
Comment (sallar) [inline src/themes.js:5]: the test text colors were white before the text rendering isn't the same on the CI vs the dev machine, so they need to be white so the tests wont fail

File: src/themes.js [hunk 5]
```diff
@@ -1,22 +1,24 @@
...
+    stroke:'rgba(27,31,35,.04)',
```
Comment (sallar) [inline src/themes.js:6]: the other themes dont have strokes... what happens?

File: src/index.js [hunk 6]
```diff
@@ -112,10 +167,12 @@ function drawYear(ctx, opts = {}) {
...
       if (moment(day.date) > today || !day.info) {
         continue;
-      }    
-      const color = theme[`grade${day.info.intensity}`];
-      ctx.fillStyle = color;
-      ctx.fillRect(
+      }
+
+      ctx.fillStyle = theme[`grade${day.info.intensity}`];
+      ctx.strokeStyle = theme["stroke"];
```
Comment (sallar) [inline src/index.js:173]: Since the other themes dont have strokes this fails

File: src/index.js [hunk 7]
```diff
@@ -161,11 +219,14 @@ function drawMetaData(ctx, opts = {}) {
...
   let themeGrades = 5;
   ctx.fillStyle = theme.text;
-  ctx.fillText('Less', width - canvasMargin - (boxWidth + boxMargin) * (themeGrades) - 55, 37);
-  ctx.fillText('More', (width - canvasMargin) - 25, 37);
+  ctx.fillText("Less", width - canvasMargin - (boxWidth + boxMargin) * (themeGrades) - 55, 37);
+  ctx.fillText("More", (width - canvasMargin) - 25, 37);
   for (let x = 0; x < 5; x += 1) {
     ctx.fillStyle = theme[`grade${x}`];
-    ctx.fillRect(width - canvasMargin - (boxWidth + boxMargin) * (themeGrades) - 27,textHeight + boxWidth,10,10);
+    ctx.strokeStyle = theme["stroke"];
```
Comment (sallar) [inline src/index.js:226]: also this

**Review by sallar — CHANGES_REQUESTED** [review]
Thanks for your PR. This is a great idea. However there are a few things that need to be fixed that I have commented on the PR.

But also, I think this needs to be a "theme option", for example the classic github theme didnt have rounded corners, etc. Or maybe an option on the whole package, that the user can control.

**Comment by xiandanin** [comment]
> Thanks for your PR. This is a great idea. However there are a few things that need to be fixed that I have commented on the PR.
> 
> But also, I think this needs to be a "theme option", for example the classic github theme didnt have rounded corners, etc. Or maybe an option on the whole package, that the user can control.

I made some modifications as you suggested:
* I added the "newGithub" theme to show that this is the latest official github style.
* Added "radius" and "stroke" parameters, but this is optional. If the theme does not contain this parameter, it will work normally.
* Change the text color of "__test__" theme to white.
* "scaleFactor" can still be passed in as a parameter, but it is dynamic by default.

**Review by xiandanin — COMMENTED** [review]
> Thanks for your PR. This is a great idea. However there are a few things that need to be fixed that I have commented on the PR.
> 
> But also, I think this needs to be a "theme option", for example the classic github theme didnt have rounded corners, etc. Or maybe an option on the whole package, that the user can control.

I made some modifications as you suggested:
* I added the "newGithub" theme to show that this is the latest official github style.
* Added "radius" and "stroke" parameters, but this is optional. If the theme does not contain this parameter, it will work normally.
* Change the text color of "__test__" theme to white.
* "scaleFactor" can still be passed in as a parameter, but it is dynamic by default.

**Comment by sallar** [comment]
I apologize for not merging this yet @xiandanin - I need to get a chance to sit down and work on this.

**Pull request closed without merging**